### ISSUE TITLE
Incorrect validation of link fragments for duplicate headings

### ISSUE THREAD

**Issue by lukasrad02** [joshrotenberg/mdbook-lint#410]
The rules `MD051`, `CONTENT006`, and `MDBOOK006` flag link fragments of duplicate headings (with a trailing counter) as incorrect, even if they are correct given the output of `mdbook`.

Given the following markdown file

```md
# Demo

## Topic

Some text about a topic. You might also want to look at its [details](#details).

### Details

Details about the topic.

## Another Topic

Some text about *another* topic. You might also want to look at its [details](#details-1).

### Details

Details about the *other* topic.

## Topic With File Name

See [details](./demo.md#details-1) above.
```

the first two links are correct, since there are two "Details" headings, which get the ids `details` and `details-1`. Clicking the second link in the rendered HTML does jump to the second heading (the chapter is rather short, so one might not see the page actually scrolling, but the heading is prepended with the `»` focus indicator). However, the second link is flagged as incorrect by `MD051` and `CONTENT006`:

```
error[MD051]: Link fragment 'details-1' is not valid
  --> src/demo.md:13:69
     |
  13 | Some text about *another* topic. You might also want to look at its [details](#details-1).
     |                                                                     ^^^^^^^^^^^^^^^^^^^^^^ link-fragments

warning[CONTENT006]: Broken internal link: '#details-1' does not match any heading in this document. Did you mean '#details'? (link text: 'details')
  --> src/demo.md:13:69
     |
  13 | Some text about *another* topic. You might also want to look at its [details](#details-1).
     |                                                                     ^^^^^^^^^^^^^^^^^^^^^^ no-broken-internal-links
```

The rule `MDBOOK006` shows the same behavior, but only checks links that include a file name, such as in the third link:

```
error[MDBOOK006]: Cross-reference anchor 'details-1' not found in './demo.md'. Did you mean 'details'?
  --> src/demo.md:21:5
     |
  21 | See [details](./demo.md#details-1) above.
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ internal-cross-references
```
